src/nspawn/nspawn.c

```diff
@@ -3486,6 +3487,16 @@ static int inner_child(
                 n_env++;
         }
 
+        if (arg_start_mode != START_BOOT) {
+                /* If we're running a command in the container, let's default to the C.UTF-8 locale as it's
+                 * part of glibc these days and was backported to most distros a long time before it got
+                 * added to upstream glibc. */
+                envp[n_env] = strdup("LANG=C.UTF-8");
```

Comment:
Shouldn't we use `"LANG=" SYSTEMD_DEFAULT_LOCALE` instead?

Like this:
https://github.com/systemd/systemd/blob/492f9e74ff2b7b07d0999a7241c16ed5e767fdeb/src/core/locale-setup.c#L77

Reply:
Made a follow up with this change in #23252.